crates/typst/src/eval/call.rs

```diff
@@ -368,6 +285,60 @@ pub(crate) fn call_closure(
     Ok(output)
 }
 
+/// Evaluate the callee and its arguments.
+///
+/// Prioritize associated functions on the value's type (i.e., methods) over its fields.
+/// A function call on a field is only allowed for functions, types, modules (because
+/// they are scopes), and symbols (because they have modifiers).
+///
+/// For dictionaries, it is not allowed because it would be ambiguous (prioritizing
+/// associated functions would make an addition of a new associated function a breaking
+/// change and prioritizing fields would break associated functions for certain
+/// dictionaries).
+fn eval_callee_args(
+    target: Value,
+    target_expr_span: Span,
+    field: Ident,
+    mut args: Args,
+) -> SourceResult<(Value, Args)> {
+    if let Some(callee) = target.ty().scope().get(&field) {
+        args.push_front(target_expr_span, target);
+        Ok((callee.clone(), args))
+    } else if matches!(
+        target,
+        Value::Symbol(_) | Value::Func(_) | Value::Type(_) | Value::Module(_)
+    ) {
+        Ok((target.field(&field).at(field.span())?, args))
+    } else {
+        let mut error = error!(
+            field.span(),
+            "type {} has no method `{}`",
+            target.ty(),
+            field.as_str(),
+        );
+
+        match target {
+            Value::Dict(ref dict) if matches!(dict.get(&field), Ok(Value::Func(_))) => {
+                error.hint(eco_format!(
+                    "to call the function stored in the dictionary, \
+                        surround the field access with parentheses, e.g. `(dict.{})(..)`",
+                    field.as_str(),
+                ));
+            }
+            _ if target.field(&field).is_ok() => {
+                error.hint(eco_format!(
+                    "did you mean to access the field `{}`?",
+                    field.as_str()
+                ));
+            }
+            _ => (),
```

Comment:
very nitpicky nitpick but i think this is more usual

```suggestion
            _ => {}
```